copy.js

```diff
@@ -0,0 +1,79 @@
+"use strict";
+
+/*
+  copy text to clipboard
+  ref: https://github.com/mdn/webextensions-examples/tree/master/context-menu-copy-link-with-types
+
+*/
+
+var global_link_text = '';
+
+function maybe_clean (orig_url, regexp, f) {
+    var res = '';
+    for (var pat in regexp) {
+        if (new RegExp(pat).test(orig_url)) {
```

Comment:
btw, the patterns in `cleaning_rules.js` aren't regexes, they're just glob-style wildcards.

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Match_patterns Looks like we're doing this. Do you know if there's a replacement to this deprecated library https://developer.mozilla.org/en-US/docs/Archive/Add-ons/Add-on_SDK/Low-Level_APIs/util_match-pattern ? We might be able to just use the same thing that Firefox is already using.

Reply:
Just out of curiosity I have investigated a bit. Unfortunately the `MatchPattern` API is only usable when hooked to an event listener:
https://developer.mozilla.org/en-US/docs/Mozilla/JavaScript_code_modules/WebRequest.jsm

WebExtension have to rely on the plain JS API